### PR TITLE
Add check for nil objects; add handler if not S3OriginConfig

### PR DESCRIPTION
**Pull request by haooliveira84**
### Description of your changes

Add check of existing `nil` objects 
Add parameter for `CloudFront Distributions` if not have `S3OriginConfig`

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### CODE REVIEW

File: converters/provider-aws/dynamodb/dynamodb.go [hunk 1]
```diff
@@ -47,35 +47,40 @@ func TableResource(mg resource.Managed) ([]resource.Managed, error) {
...
-		}
-		for _, a := range t.KeySchema {
-			if *a.KeyType == "HASH" {
-				parameter.HashKey = a.AttributeName
-			}
-			if *a.KeyType == "RANGE" {
-				parameter.RangeKey = a.AttributeName
+	if source.Spec.ForProvider.GlobalSecondaryIndexes != nil {
+		for _, t := range source.Spec.ForProvider.GlobalSecondaryIndexes {
+			if t.IndexName == nil {
```
Comment (sergenyalcin) [inline converters/provider-aws/dynamodb/dynamodb.go:52]: I am asking only to understand and ensure we are doing the right conversion. Is this check the correct one? Now, we are doing the assignment process if the IndexName is nil. Should we do this in this case, or should we do this if it is not nil?

**Review by sergenyalcin — COMMENTED** [review]
Thanks @haooliveira84 I left a comment to ensure that we are doing the right conversion for dynamodb resource

File: converters/provider-aws/dynamodb/dynamodb.go (hunk 1 above)
Comment (haooliveira84) [reply converters/provider-aws/dynamodb/dynamodb.go]: my bad @sergenyalcin, I already fixed it.

File: converters/provider-aws/cloudfront/distribution.go [hunk 2]
```diff
@@ -50,13 +50,34 @@ func DistributionResource(mg resource.Managed) ([]resource.Managed, error) {
 	target.Spec.ForProvider.ViewerCertificate[0].MinimumProtocolVersion = source.Spec.ForProvider.DistributionConfig.ViewerCertificate.MinimumProtocolVersion
 
 	target.Spec.ForProvider.WebACLID = source.Spec.ForProvider.DistributionConfig.WebACLID
+	if source.Spec.ForProvider.DistributionConfig.Origins.Items != nil {
```
Comment (sergenyalcin) [inline converters/provider-aws/cloudfront/distribution.go:53]: If the length of the `Items` slice is 0, then this for loop will be skipped. So, I am not sure that we really need this nil-check. However, the `Origins` is a pointer in the API definition and can be nil. In this case, we must add a nil check for `Origins` here.

File: converters/provider-aws/dynamodb/dynamodb.go [hunk 3]
```diff
@@ -47,44 +47,49 @@ func TableResource(mg resource.Managed) ([]resource.Managed, error) {
...
-			NonKeyAttributes: t.Projection.NonKeyAttributes,
-			ProjectionType:   t.Projection.ProjectionType,
-		}
-		for _, a := range t.KeySchema {
-			if *a.KeyType == "HASH" {
-				parameter.HashKey = a.AttributeName
-			}
-			if *a.KeyType == "RANGE" {
-				parameter.RangeKey = a.AttributeName
+	if source.Spec.ForProvider.GlobalSecondaryIndexes != nil {
```
Comment (sergenyalcin) [inline converters/provider-aws/dynamodb/dynamodb.go:50]: Do we really need this check for this slice?

File: converters/provider-aws/dynamodb/dynamodb.go [hunk 4]
```diff
@@ -47,44 +47,49 @@ func TableResource(mg resource.Managed) ([]resource.Managed, error) {
...
-	for _, t := range source.Spec.ForProvider.LocalSecondaryIndexes {
-		parameter := &targetv1beta1.LocalSecondaryIndexParameters{
-			Name:             t.IndexName,
-			NonKeyAttributes: t.Projection.NonKeyAttributes,
-			ProjectionType:   t.Projection.ProjectionType,
-		}
-		for _, a := range t.KeySchema {
-			if *a.KeyType == "RANGE" {
-				parameter.RangeKey = a.AttributeName
+	if source.Spec.ForProvider.LocalSecondaryIndexes != nil {
```
Comment (sergenyalcin) [inline converters/provider-aws/dynamodb/dynamodb.go:70]: Do we really need this check for this slice?

**Review by sergenyalcin — COMMENTED** [review]
Thanks @haooliveira84 I left a few comments.

File: converters/provider-aws/cloudfront/distribution.go (hunk 2 above)
Comment (haooliveira84) [reply converters/provider-aws/cloudfront/distribution.go:53]: I'll fix

File: converters/provider-aws/dynamodb/dynamodb.go (hunk 3 above)
Comment (haooliveira84) [reply converters/provider-aws/dynamodb/dynamodb.go:50]: @sergenyalcin this slice can be null?

File: converters/provider-aws/dynamodb/dynamodb.go (hunk 4 above)
Comment (haooliveira84) [reply converters/provider-aws/dynamodb/dynamodb.go:70]: @sergenyalcin this slice can be null?

**Review by sergenyalcin — APPROVED** [review]
Thanks @haooliveira84 LGTM!

**Pull request merged**